code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/EventHub.kt

```diff
@@ -173,15 +214,45 @@ internal class EventHub(val eventHistory: EventHistory?) {
     /**
      * `EventHub` will begin processing `Event`s when this API is invoked.
      */
-    fun start() {
+    @JvmOverloads
+    fun start(completion: (() -> Unit)? = null) {
         eventHubExecutor.submit {
-            this.hubStarted = true
-            this.eventDispatcher.start()
-            this.shareEventHubSharedState()
-            Log.trace(CoreConstants.LOG_TAG, LOG_TAG, "Event Hub successfully started")
+            if (hubStartReceived) {
+                Log.debug(CoreConstants.LOG_TAG, LOG_TAG, "Dropping start call as it was already received")
+                return@submit
+            }
+
+            this.hubStartReceived = true
+            this.hubStartCallback = completion
+
+            tryStartHub()
         }
     }
 
+    private fun tryStartHub() {
```

Comment:
This method is being called from 2 different places 
- `start()`  invoked on `eventHubExecutor` thread
- `extensionPostRegistration()` invoked on `extensionInitExecutor` thread.

If both the invocations execute simultaneously (say for the last extension registration followed by mobile core start in customer app code) then we may end up in a situation where the hub is never started.
Wonder if we should  make the boolean gates to be volatile or serialize these operations here.

Reply:
`extensionPostRegistration` is also invoked on `eventHubExecutor`.
https://github.com/praveek/aepsdk-core-android/blob/dev-v2.0.0/code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/EventHub.kt#L316

Reply:
Ah! My bad. Misread it. LGTM.